tests/unit/s2n_self_talk_session_resumption_test.c

```diff
@@ -531,6 +534,128 @@ int main(int argc, char **argv)
+        EXPECT_SUCCESS(s2n_connection_get_session(client_conn, tls12_ticket.blob.data, tls12_ticket_length));
+
+        struct s2n_config *client_config[] = { tls12_client_config, tls13_client_config };
+
+        /* A quirk of the TLS1.2 session resumption behavior is that if a ticket is set
+         * on the connection using s2n_connection_set_session, s2n_connection_get_session 
+         * will return a valid ticket, even before actually receiving a new session ticket
+         * from the server. Here we test that behavior to ensure it is consistant. */
+        {
+            uint8_t *session_state = NULL;
+
+            /* Prepare client and server for new connection */
+            EXPECT_SUCCESS(s2n_connection_wipe(client_conn));
+            EXPECT_SUCCESS(s2n_connection_wipe(server_conn));
+            EXPECT_SUCCESS(s2n_connections_set_io_pair(client_conn, server_conn, &io_pair));
+
+            EXPECT_SUCCESS(s2n_connection_set_config(client_conn, tls12_client_config));
+
+            /* Client sets up a resumption connection with the received session ticket data */
+            EXPECT_SUCCESS(s2n_connection_set_session(client_conn, tls12_ticket.blob.data, s2n_stuffer_data_available(&tls12_ticket)));
+
+            /* s2n_connection_get_session will be non-zero if a TLS1.2 ticket was set on the connection */
+            uint32_t session_length = s2n_connection_get_session_length(client_conn);
+            EXPECT_TRUE(session_length > 0);
+
+            /* Call get_session to retreive session ticket */
+            session_state = realloc(session_state, session_length);
+            EXPECT_NOT_NULL(session_state);
+            EXPECT_SUCCESS(s2n_connection_get_session(client_conn, session_state, session_length));
+
+            /* Check that the session ticket returned is valid by deserializing it */
+            struct s2n_blob blob = { 0 };
+            EXPECT_SUCCESS(s2n_blob_init(&blob, session_state, session_length));
+            struct s2n_stuffer session_state_stuffer = { 0 };
+            EXPECT_SUCCESS(s2n_stuffer_init(&session_state_stuffer, &blob));
+            EXPECT_SUCCESS(s2n_stuffer_skip_write(&session_state_stuffer, session_length));
+            EXPECT_SUCCESS(s2n_client_deserialize_resumption_state(client_conn, &session_state_stuffer));
+
+            /* Negotiate TLS1.2 connection */
+            EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_conn, client_conn));
+
+            /* The session length should be non-zero after a client has received a session ticket */
+            EXPECT_TRUE(s2n_connection_get_session_length(client_conn) > 0);
+
+            free(session_state);
+        }
+
+        /* Tests that if a TLS1.3 ticket is set on the connection, s2n_connection_get_session will
+         * not return a ticket until a session ticket is sent by the server as a post-handshake
+         * message. */
+        {
+            for (size_t j = 0; j < s2n_array_len(client_config); j++) {
```

Comment:
Nit: you probably don't need both a for loop scope and an empty scope. The for loop + comment should be enough to mark the test as separate.